runtime error: invalid memory address or nil pointer dereference

Hi,

I'm seeing the below using gcr.io/gcp-runtimes/container-structure-test:**v0.1.3** with the docker driver:


```
Unable to find image 'gcr.io/gcp-runtimes/container-structure-test:v0.1.3' locally
v0.1.3: Pulling from gcp-runtimes/container-structure-test
cd2f5b7886e9: Already exists
9be56f236f9a: Pulling fs layer
9be56f236f9a: Verifying Checksum
9be56f236f9a: Download complete
9be56f236f9a: Pull complete
Digest: sha256:15ae6f3996ebc05b657a042bd5dd8f4d4d66b015effea5fd024c70b8a1f9c42a
Status: Downloaded newer image for gcr.io/gcp-runtimes/container-structure-test:v0.1.3
Using driver docker
=== RUN   TestAll
2018/02/12 15:53:06 Running tests for file tests/container_tests.yaml
=== RUN   TestAll/Command_Test:_Nginx_version_is_correct
=== RUN   TestAll/Command_Test:_Nginx_config_test_passes
=== RUN   TestAll/Command_Test:_Can_bind_to_low_ports
=== RUN   TestAll/File_Existence_Test:_Nginx_config_exists
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x769b58]

goroutine 26 [running]:
testing.tRunner.func1(0xc42005d450)
	/builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/go1_8_3_linux_amd64/src/testing/testing.go:622 +0x29d
panic(0x7d1d60, 0xa04db0)
	/builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/go1_8_3_linux_amd64/src/runtime/panic.go:489 +0x2cf
github.com/GoogleCloudPlatform/container-structure-test/types/v2.(*StructureTest).RunFileExistenceTests.func1(0xc42005d450)
	types/v2/structure.go:101 +0x278
testing.tRunner(0xc42005d450, 0xc4201b9220)
	/builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/go1_8_3_linux_amd64/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/external/go1_8_3_linux_amd64/src/testing/testing.go:697 +0x2ca
```

If I remove all fileExistenceTests I do not see the error. Test yaml is below:

```yaml
schemaVersion: 2.0.0

globalEnvVars:
  - key: 'UPSTREAM'
    value: '127.0.0.1:8080'

fileExistenceTests:
- name: 'Nginx config exists'
  path: '/etc/nginx/nginx.conf'
  shouldExist: true
  permissions: '-rw-r--r--'
- name: 'Proxy config exists'
  path: '/etc/nginx/conf.d/proxy.conf'
  shouldExist: true
  permissions: '-rw-r--r--'
- name: 'Certs directory exists'
  path: '/etc/nginx/certs'
  shouldExist: true
  permissions: 'drwxr-xr-x'
- name: 'Pid directory exists'
  path: '/var/run/nginx'
  shouldExist: true
  permissions: 'drwxr-xr-x'
- name: 'Setup script exists'
  path: '/usr/local/bin/setup.sh'
  shouldExist: true
  permissions: '-rwxr-xr-x'
- name: 'Start script exists'
  path: '/usr/local/bin/start.sh'
  shouldExist: true
  permissions: '-rwxr-xr-x'

commandTests:
- name: 'Nginx version is correct'
  command: 'nginx'
  args: ['-v']
  expectedError: ['1.13.7']
- name: 'Nginx config test passes'
  command: 'nginx'
  setup: [['/usr/local/bin/setup.sh']]
  args: ['-t', '-c', '/etc/nginx/nginx.conf']
  exitCode: '0'
- name: 'Can bind to low ports'
  command: 'getcap'
  args: ['/usr/sbin/nginx']
  expectedOutput: ['cap_net_bind_service\+ep']

fileContentTests:
- name: 'Correct directives in nginx.conf'
  path: '/etc/nginx/nginx.conf'
  expectedContents: [
    'user nginx;',
    'pid /var/run/nginx/nginx.pid',
    'include /etc/nginx/conf.d/\*.conf'
  ]
- name: 'Correct redirects & proxying'
  path: '/etc/nginx/conf.d/proxy.conf'
  expectedContents: [
    'listen 443 ssl',
    'return  301 https://\$host\$request_uri',
    'proxy_pass          http://\$UPSTREAM'
  ]

metadataTest:
  exposedPorts: ['80', '443']
  cmd: ['/usr/local/bin/start.sh']
```

Any help is appreciated

Cheers,
Mike

--- a/types/v2/structure.go
+++ b/types/v2/structure.go
@@ -91,14 +91,14 @@ func (st *StructureTest) RunFileExistenceTests(t *testing.T) int {
 			var info os.FileInfo
 			info, err = driver.StatFile(t, tt.Path)
 			if tt.ShouldExist && err != nil {
-				t.Errorf("File %s should exist but does not, got error: %s!", tt.Path, err)
+				t.Fatalf("File %s should exist but does not, got error: %s!", tt.Path, err)
 			} else if !tt.ShouldExist && err == nil {
-				t.Errorf("File %s should not exist but does!", tt.Path)
+				t.Fatalf("File %s should not exist but does!", tt.Path)
 			}
 			if tt.Permissions != "" {
 				perms := info.Mode()
 				if perms.String() != tt.Permissions {
-					t.Errorf("%s has incorrect permissions. Expected: %s, Actual: %s", tt.Path, tt.Permissions, perms.String())
+					t.Fatalf("%s has incorrect permissions. Expected: %s, Actual: %s", tt.Path, tt.Permissions, perms.String())
 				}
 			}
 			counter++
@@ -119,7 +119,7 @@ func (st *StructureTest) RunFileContentTests(t *testing.T) int {
 			defer driver.Destroy(t)
 			actualContents, err := driver.ReadFile(t, tt.Path)
 			if err != nil {
-				t.Errorf("Failed to open %s. Error: %s", tt.Path, err)
+				t.Fatalf("Failed to open %s. Error: %s", tt.Path, err)
 			}
 
 			contents := string(actualContents[:])

